homeassistant/components/kef/media_player.py

```diff
@@ -304,3 +398,73 @@ def icon(self):
     async def async_media_next_track(self):
         """Send next track command."""
         await self._speaker.next_track()
+
+    async def update_dsp(self) -> None:
+        """Update the DSP settings."""
+        if self._speaker_type == "LS50" and self._state == STATE_OFF:
+            # The LSX is able to respond when off the LS50 has to be on.
+            return
+
+        (mode, *rest) = await asyncio.gather(
+            self._speaker.get_mode(),
+            self._speaker.get_desk_db(),
+            self._speaker.get_wall_db(),
+            self._speaker.get_treble_db(),
+            self._speaker.get_high_hz(),
+            self._speaker.get_low_hz(),
+            self._speaker.get_sub_db(),
+        )
+        keys = ["desk_db", "wall_db", "treble_db", "high_hz", "low_hz", "sub_db"]
+        self._dsp = dict(zip(keys, rest), **mode._asdict())
+
+    async def async_added_to_hass(self):
+        """Update the DSP settings after adding to HASS."""
+        await self.update_dsp()
```

Comment:
This doesn't seem necessary because you already have set `update_before_add=True`, and you update dsp inside async_update.

Reply:
Fixed